src/components/TableComponent/TableComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import './TableComponent.scss';
+import { TableRow } from './index';
+import { SearchTerms } from '../../types/types';
+import SearchIcon from '../../assets/icons/Search.svg';
+
+interface TableProps {
+  data: Array<Record<string, any>>;
+}
+
+const TableComponent: React.FC<TableProps> = ({ data }) => {
+  const [filteredData, setFilteredData] = useState(data);
+  const [searchTerms, setSearchTerms] = useState<Record<string, string>>(
+    Object.keys(data[0] || {}).reduce((acc, key) => {
+      acc[key] = '';
+      return acc;
+    }, {} as Record<string, string>)
+  );
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>, column: string) => {
+    const newSearchTerms = { ...searchTerms, [column]: e.target.value };
+    setSearchTerms(newSearchTerms);
+    filterData(newSearchTerms);
+  };
+
+  const filterData = (searchTerms: Record<string, string>) => {
+    const filtered = data.filter(row =>
+      Object.keys(searchTerms).every(
+        key => String(row[key]).toLowerCase().includes(searchTerms[key].toLowerCase())
+      )
+    );
+    setFilteredData(filtered);
+  };
+
+  return (
+    <div className="tableWrapper">
+      <div className="tableHeader">
+        {Object.keys(data[0] || {}).map((key, index) => (
+          <div key={index} className={`tableHeader-cell ${data[0][key]?.tableCellType}`}>
```

Comment:
Let's rename `tableCellType` to `cellType`. Since this is used in a table, it should be clear that we are referring to table cells.